pkg/apiutil/apiutil.go

```diff
@@ -127,3 +132,56 @@ func ErrorResp(rd *render.Render, w http.ResponseWriter, err error) {
 		rd.JSON(w, http.StatusInternalServerError, err.Error())
 	}
 }
+
+// GetIPAddrFromGRPCContext returns gRPC client IP from context
+func GetIPAddrFromGRPCContext(ctx context.Context) string {
+	ip, ok := GetRealIPAddrFromGRPCContext(ctx)
+	if ok {
+		return ip
+	}
+	ip = GetPeerAddrFromGRPCContext(ctx)
+	return ip
+}
+
+func GetRealIPAddrFromGRPCContext(ctx context.Context) (string, bool) {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return "", ok
+	}
+	realIPs := md.Get("x-real-ip")
+	if len(realIPs) == 0 {
+		return "", false
+	}
+	return realIPs[0], true
+}
+
+func GetPeerAddrFromGRPCContext(ctx context.Context) string {
+	var addr string
+	if pr, ok := peer.FromContext(ctx); ok {
+		if tcpAddr, ok := pr.Addr.(*net.TCPAddr); ok {
+			addr = tcpAddr.IP.String()
+		} else {
+			addr = pr.Addr.String()
+		}
+	}
+	return addr
+}
+
+// GetIPAddrFromHTTPRequest returns http client IP from context
+func GetIPAddrFromHTTPRequest(r *http.Request) string {
+	ips := strings.Split(r.Header.Get("X-Forwarded-For"), ",")
+	if ips[0] != "" {
+		return ips[0]
+	}
+
+	ip := r.Header.Get("X-Real-Ip")
```

Comment:
Can we make it consistant? I see `x-real-ip` in `GetRealIPAddrFromGRPCContext`.

Reply:
ok, http also supports lowercase